### PR TITLE
Unregister requests when they timeout to adapt to Lunchbox/#233.

### PR DESCRIPTION
**Pull request by hernando**
I've grepped the source for "wait(" and "waitRequest(". I've fixed the code where wait appears with a timeout to unregister the request after the timeout and rethrow if needed; waitRequest never appears with a timeout.

### CODE REVIEW

**Comment by hernando** [comment]
Not building because it depends on Eyescale/Lunchbox#233

**Comment by hernando** [comment]
@tribal-tec do you think you can review this?

**Comment by tribal-tec** [comment]
Looks good to me. I'm a bit confused what @eile did for Equalizer, but in conjunction with the Lunchbox change this change here seems right. But I'm not comfortable merging it w/o having a statement from @eile.

**Comment by eile** [comment]
Ok, I'm back somewhat.

Eyescale/Lunchbox#233 increases the complexity substantially, while adding a little more flexibility. I did already write some comments in https://github.com/Eyescale/Lunchbox/commit/274c2445b57b1fe78b685fd49d3aef827b583ca7#commitcomment-13851199

Skype?

**Comment by hernando** [comment]
@tribal-tec please review is you can.

File: co/objectStore.cpp [hunk 1]
```diff
@@ -206,13 +206,21 @@ NodeID ObjectStore::findMasterNodeID( const uint128_t& identifier )
...
 
-        if( masterNodeID != 0 )
+            if( masterNodeID != 0 )
+            {
+                LBLOG( LOG_OBJECTS ) << "Found " << identifier << " on "
+                                     << masterNodeID << std::endl;
+                return masterNodeID;
+            }
+        }
+        catch( lunchbox::FutureTimeout& )
```
Comment (eile) [inline co/objectStore.cpp]: const

**Comment by eile** [comment]
+1 otherwise

**Comment by tribal-tec** [comment]
+1

**Pull request merged**